graphrag/llm/openai/utils.py

```diff
@@ -87,17 +88,46 @@ def get_completion_llm_args(
     }
 
 
-def try_parse_json_object(input: str) -> dict:
+def try_parse_json_object(input: str) -> tuple[str, dict]:
+    """JSON cleaning and formatting utilities."""
+    """sometime, the llm return a json string with some extra description, this function will clean it up."""
+    _pattern = r"\{(.*)\}"
+    _match = re.search(_pattern, input)
+    input = "{" + _match.group(1) + "}" if _match else input
+    
+    """Clean up json string."""
+    input = (
+        input.replace("{{","{")
+        .replace("}}","}")
+        .replace('"[{', "[{")
+        .replace('}]"', "}]")
+        .replace("\\", " ")
+        .replace("\\n", " ")
+        .replace("\n", " ")
+        .replace("\r", "")
+        .strip()
+    )
+
+    # Remove JSON Markdown Frame
+    if input.startswith("```json"):
+        input = input[len("```json") :]
+    if input.endswith("```"):
+        input = input[: len(input) - len("```")]
+
+    """Fixup potentially malformed json string using json_repair."""
+    input = str(repair_json(json_str=input, return_objects=False))
```

Comment:
Before calling out to another library via `repair_json` to fix what could be a valid json string at this point, can you add in an attempt to call `json.loads(json_str=input)` before this line...and only call `repair_json()` if the string is still invalid? String operations can be slow at large scale so we want to minimize them where possible.

Reply:
add try json.loads() block before repair_json
Many thanks @jgbradley1